tgui/packages/tgui/interfaces/ForceEvent.js

```diff
@@ -0,0 +1,124 @@
+import { useBackend, useLocalState } from '../backend';
+import { Stack, Button, Icon, Input, Section, Table } from '../components';
+import { Window } from '../layouts';
+import { flow } from 'common/fp';
+import { filter, sortBy } from 'common/collections';
+
+export const ForceEvent = (props, context) => {
+  return (
+    <Window title="Force Event" width={450} height={450}>
+      <Window.Content scrollable>
+        <EventSearch />
+        <EventOptionsPanel />
+        <EventContent />
+      </Window.Content>
+    </Window>
+  );
+};
+
+export const EventSearch = (props, context) => {
+  const [searchQuery, setSearchQuery] = useLocalState(
+    context,
+    'searchQuery',
+    ''
+  );
+
+  return (
+    <Section>
+      <Stack>
+        <Stack.Item>
+          <Icon name="search" />
+        </Stack.Item>
+        <Stack.Item grow>
+          <Input
+            autoFocus
+            fluid
+            onInput={(e) => setSearchQuery(e.target.value)}
+            placeholder="Search..."
+            value={searchQuery}
+          />
+        </Stack.Item>
+      </Stack>
+    </Section>
+  );
+};
+
+export const EventOptionsPanel = (props, context) => {
+  const { data } = useBackend(context);
+
+  const [announce, setAnnounce] = useLocalState(context, 'announce', true);
+
+  return (
+    <Button.Checkbox
+      fluid
+      checked={announce}
+      onClick={() => setAnnounce(!announce)}>
+      Announce event to the crew
+    </Button.Checkbox>
+  );
+};
+
+export const EventContent = (props, context) => {
+  const { data } = useBackend(context);
+
+  const categories = Object.values(data.categories);
+  const sortCategories = sortBy((category) => category.name);
+
+  return (
+    <Section>
```

Comment:
I don't think you need this section, and you don't need a vertical stack there either. In the map, directly return EventList elements.

Reply:
Thank you, I have done so.